server/opts.go

```diff
@@ -2885,6 +2885,15 @@ func overrideCluster(opts *Options) error {
 		opts.Cluster.Port = 0
 		return nil
 	}
+	// -1 will fail url.Parse, so if we have -1, change it to
+	// 0, and then on parse, replace the port with -1 so we get
+	// automatic port allocation
+	wantsRandom := false
+	if strings.HasSuffix(opts.Cluster.ListenStr, ":-1") {
```

Comment:
I have checked and we don't use `opts.Cluster.ListenStr` anywhere in the server code (outside of command line parsing I mean), so I think it is ok. Otherwise, I would have suggested to do `listenStr := opts.Cluster.ListenStr` and do all the changes/reference from there on on `listenStr` so that we don't alter opts.Cluster.ListenStr.